providers/Flipt/tests/FliptProviderTest.php

```diff
@@ -0,0 +1,147 @@
+
+        $result = $this->provider->resolveIntegerValue( 'flag', 10, new EvaluationContext( 'id', new Attributes( [ 'context' => 'demo' ] ) ) );
+
+        $this->assertInstanceOf( ResolutionDetails::class, $result );
+        $this->assertEquals( $result->getValue(), 20 );
+
+    }
+
+    public function testFloat() 
+    {
+        $this->mockClient->shouldReceive( 'variant')
+            ->withArgs( function( $flag, $contextRecv, $entityId ) {
+                $this->assertEquals( $flag, 'flag' );
+                $this->assertEquals( $contextRecv, [ 'context' => 'demo' ] );
+                $this->assertEquals( $entityId, 'id' );
+                return true;
+            })
+            ->andReturn( new DefaultVariantEvaluationResult( true, 'MATCH_EVALUATION_REASON', 0.1, 'rid', '13245', [], '0.2345', '{"json":1}' ) );
+
+        $result = $this->provider->resolveFloatValue( 'flag', 0.1111, new EvaluationContext( 'id', new Attributes( [ 'context' => 'demo' ] ) ) );
+
+        $this->assertInstanceOf( ResolutionDetails::class, $result );
+        $this->assertEquals( $result->getValue(), 0.2345 );
+
+    }
+
+    public function testString() 
+    {
+        $this->mockClient->shouldReceive( 'variant')
+            ->withArgs( function( $flag, $contextRecv, $entityId ) {
+                $this->assertEquals( $flag, 'flag' );
+                $this->assertEquals( $contextRecv, [ 'context' => 'demo' ] );
+                $this->assertEquals( $entityId, 'id' );
+                return true;
+            })
+            ->andReturn( new DefaultVariantEvaluationResult( true, 'MATCH_EVALUATION_REASON', 0.1, 'rid', '13245', [], 'My string', '{"json":1}' ) );
+
+        $result = $this->provider->resolveStringValue( 'flag', 'base', new EvaluationContext( 'id', new Attributes( [ 'context' => 'demo' ] ) ) );
+
+        $this->assertInstanceOf( ResolutionDetails::class, $result );
+        $this->assertEquals( $result->getValue(), 'My string' );
+
+    }
+
+
+    public function testObject() 
+    {
+        $this->mockClient->shouldReceive( 'variant')
+            ->withArgs( function( $flag, $contextRecv, $entityId ) {
+                $this->assertEquals( $flag, 'flag' );
+                $this->assertEquals( $contextRecv, [ 'context' => 'demo' ] );
+                $this->assertEquals( $entityId, 'id' );
+                return true;
+            })
+            ->andReturn( new DefaultVariantEvaluationResult( true, 'MATCH_EVALUATION_REASON', 0.1, 'rid', '13245', [], 'My string', '{"json":1}' ) );
+
+        $result = $this->provider->resolveObjectValue( 'flag', [], new EvaluationContext( 'id', new Attributes( [ 'context' => 'demo' ] ) ) );
+
+        $this->assertInstanceOf( ResolutionDetails::class, $result );
+        $this->assertEquals( $result->getValue(), [ "json" => 1 ] );
+
+    }
+
+
+    public function testCache()
```

Comment:
I would suggest having test files per class. The cache being separate, it should exist in a `CacheTest.php` separate from the provider tests

Reply:
I refactored the whole Caching mechanism into its own CacheProvider.
This is now a wrapper around other Providers to allow caching. This way the FliptProvider does not know anything about caching und could be an approach for https://github.com/open-feature/php-sdk/issues/106.

I also remove the tests in it own file.